src/backend/core/tests/test_api_users.py

```diff
@@ -440,6 +440,56 @@ def test_api_users_list_pagination(
     assert len(content["results"]) == 2
 
 
+@pytest.mark.parametrize("page_size", [1, 10, 99, 100])
+def test_api_users_list_pagination_page_size(
+    page_size,
+):
+    """Page's size on pagination should work as expected."""
+    identity = factories.IdentityFactory()
+    user = identity.user
+
+    client = APIClient()
+    client.force_login(user)
+
+    factories.UserFactory.create_batch(page_size)
+
+    response = client.get(
+        f"/api/v1.0/users/?page_size={page_size}",
+    )
+
+    assert response.status_code == HTTP_200_OK
+    content = response.json()
+
+    assert content["count"] == page_size + 1
+    assert len(content["results"]) == page_size
```

Comment:
I feel like you could have used the test just above, WDYT ?

Reply:
Actually maybe, but test just above mock page_size, so how could set it through a query parameter? The test just above actually tests the pagination mechanism, more than the page size. I preferred having simpler tests, that tests a well-scoped unit of code. However that's a personal opinion!

Reply:
+1 on dedicated tests with a single goal.